ddtrace/tracer/metrics_test.go

```diff
@@ -284,8 +284,10 @@ func TestReportHealthMetrics(t *testing.T) {
 		flushChan:        make(chan chan<- struct{}),
 		exitChan:         make(chan struct{}),
 		payloadChan:      make(chan []*span, payloadQueueSize),
+		sendChan:         make(chan *payload, payloadQueueSize),
 		stopped:          make(chan struct{}),
 		prioritySampling: newPrioritySampler(),
+		payloads:         newPool(),
```

Comment:
You can do this if you like, but we must prove that it provides an actual benefit. I think it's a bit overkill but you can write a benchmark and post the numbers with a pool and without.

Reply:
Generally pools are declared at the global level ([example](https://github.com/golang/go/blob/a570fcf74bdf652f453ed5dc0d8a41b15f1af8de/src/net/http/header.go#L156)) to really bring out their fullest: obviously concurrent and available across all instances the the object (in our case tracer).

Reply:
I think that's a good idea. The pool makes sense if the object is spawned very often, but that might not be the case for most of the uses. We can explore it later if it ever turns out to be a bottleneck.